### PR TITLE
Feature/fill locale

### PR DESCRIPTION
**Pull request by israelst**
Better merged after #5.

The memoization technique was used to prevent unnecessary io, which significantly improved the performance. 

Is there some special reason to have an id other than the geocode in `Localidade` table?

### CODE REVIEW

**Comment by fccoelho** [comment]
@israelst , the reason we use an id, is because there are cases in which the localidade is not an official IBGE polygon, and so it doesn't have a gecode.

I notice you propose a refactoring of some of the utilities  scripts, but does not remove the originals, could you elaborate on why?

**Comment by israelst** [comment]
@fccoelho what should be the value of the id? Auto increment?

I didn't remove it because i don't know if the old db is already deprecated.

**Comment by fccoelho** [comment]
I use the type bigserial from postgresql for the id, so yes, it is auto
increment.



On Wed, Oct 7, 2015 at 4:47 PM, Israel Teixeira <notifications@github.com>
wrote:

> @fccoelho <https://github.com/fccoelho> what should be the value of the
> id? Auto increment?
>
> I didn't remove it because i don't know if the old db is already
> deprecated.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/fccoelho/AlertaDengueCaptura/pull/8#issuecomment-146308341>
> .
>



-- 
Flávio Codeço Coelho
================
+55(21) 3799-5551
Professor
Escola de Matemática Aplicada
Fundação Getulio Vargas
Praia de Botafogo, 190 sala 312
Rio de Janeiro - RJ
22250-900
Brasil

**Comment by israelst** [comment]
@fccoelho, could you please review these new commits?

**Comment by israelst** [comment]
BTW, I'm assuming that the overlap between  `"Dengue_global"."Municipio"` and `"Municipio"."Localidade"` is intentional.

Sorry by my lack of understanding of the schema.

File: dbschema/Banco_de_dados_do_Alerta_Dengue_create.sql [hunk 1]
```diff
@@ -206,7 +205,7 @@ CREATE TABLE "Municipio"."Ovitrampa" (
 
 -- Table: "Municipio"."Tweet"
 CREATE TABLE "Municipio"."Tweet" (
-    "id" BIGSERIAL PRIMARY KEY NOT NULL,
+    "id" BIGSERIAL NOT NULL,
```
Comment (fccoelho) [inline dbschema/Banco_de_dados_do_Alerta_Dengue_create.sql]: All tables must have a Primary key. so you should undo this change.

**Comment by fccoelho** [comment]
Yes Localidade is similar to Municipio because they both describe geographical areas, "Municipio"."Localidade" is a region within a  "Dengue_global"."Municipio"

File: dbschema/Banco_de_dados_do_Alerta_Dengue_create.sql (hunk 1 above)
Comment (israelst) [inline dbschema/Banco_de_dados_do_Alerta_Dengue_create.sql]: The primary key is defined [here](https://github.com/fccoelho/AlertaDengueCaptura/pull/8/files#diff-c2e28fbd9d078b9162bab8824798792eR213). It was defined twice before 1138463.
